GCC Warning comparison between signed and unsigned

GCC warnings for signed/unsigned comparisons:
```
src/MicroPather/micropather.cpp: In member function 'void micropather::PathCache::AddItem(const micropather::PathCache::Item&)':
src/MicroPather/micropather.cpp:804:13: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (index == allocated)

       ~~~~~~^~~~~~~~~~~~

src/MicroPather/micropather.cpp: In member function 'const micropather::PathCache::Item* micropather::PathCache::Find(void*, void*)':
src/MicroPather/micropather.cpp:826:13: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (index == allocated)

       ~~~~~~^~~~~~~~~~~~

src/UI/Core/ListBox.cpp: In member function 'virtual void ListBox::update()':
src/UI/Core/ListBox.cpp:302:9: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (i == mCurrentHighlight)
       ~~^~~~~~~~~~~~~~~~~~~~

src/UI/IconGrid.cpp: In member function 'void IconGrid::raiseChangedEvent()':
src/UI/IconGrid.cpp:380:24: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  if (mCurrentSelection != constants::NO_SELECTION)
      ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~

src/UI/IconGrid.cpp: In member function 'virtual void IconGrid::update()':
src/UI/IconGrid.cpp:428:24: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  if (mCurrentSelection != constants::NO_SELECTION)
      ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~

src/UI/IconGrid.cpp:438:22: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  if (mHighlightIndex != constants::NO_SELECTION)
      ~~~~~~~~~~~~~~~~^~~~~~~~~~~~

```

--- a/src/MicroPather/micropather.h
+++ b/src/MicroPather/micropather.h
@@ -349,8 +349,8 @@ namespace micropather
 		const Item* Find(void* start, void* end);
 
 		Item* mem = nullptr;
-		int allocated = 0;
-		int nItems = 0;
+		unsigned allocated = 0;
+		unsigned nItems = 0;
 	};
 
 	struct CacheData

